README.md

```diff
@@ -124,6 +124,32 @@ You can build the initramfs built by u-root into the kernel via the
 `CONFIG_INITRAMFS_SOURCE` config variable or you can load it separately via an
 option in for example Grub or the QEMU command line or coreboot config variable.
 
+## Cross Compilation (targeting different architectures and OSes)
+
+Cross-OS and -architecture compilation comes for free with Go. In fact, every PR to the u-root repo
+is built against the following architectures: amd64, x86 (i.e. 32bit), mipsle, armv7, arm64, and ppc64le.
+
+Further, we run integration tests on linux/amd64, freebsd/amd64 and linux/arm64, using several
+CI systems.
+If you need to add another CI system, processor or OS, please let us know.
+
+To cross compile for an ARM, on Linux:
+
+```shell
+GOARCH=arm u-root
+```
+
+If you are on OSX, and wish to build for Linux on AMD64:
+
+```shell
+GOOS=linux GOARCH=amd64 u-root
+```
+
+If you are on Linux, you can target FreeBSD:
```

Comment:
As of now, this would print a warning and fail to build later on: `GOOS is not linux. Did you mean to set GOOS=linux?`

That actually puzzles me; can we use a full u-root build with a FreeBSD kernel in some way? With Linux, it is fairly straightforward to just boot the default cpio with it; FreeBSD doesn't offer that AFAIK. So I would rather leave this out to not confuse people. Many commands would work of course and I could build a tarball or something to create a chroot from; although I'm not sure u-root already offers that option or that it would really work to init the OS. We should rather clarify that in a section such as "Use with non-Linux kernels" later on I think.